components/hab/src/main.rs

```diff
@@ -1021,7 +1021,7 @@ fn sub_svc_status(m: &ArgMatches<'_>) -> Result<()> {
                         None => {
                             return Err(SrvClientError::from(io::Error::from(
                                 io::ErrorKind::UnexpectedEof,
-                            )))
+                            )));
```

Comment:
Again, only suggesting this since we're changing the code anyway, but for this sort of structure, I think this approach is a bit shorter and clearer:
```rust
                    if let Some(m) = reply {
                        print_svc_status(&mut out, m, true)?;
                    } else {
                        return Err(SrvClientError::from(io::Error::from(
                            io::ErrorKind::UnexpectedEof,
                        )));
                    }
```